### PR TITLE
Core/Scripts: boss_majordomo hardcoded text

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed**:

2 more text lines removed in the ongoing series "Remove hardcoded text from core scripts".

Core script for boss Majordomo Executus:
- add missing gossip_menu_option entries
- link the options via the table gossip_menu
- remove hardcoded text from the script
- minor enum line order sorting cleanup
- replace `OnGossipSelect()` with `sGossipSelect()`

Please note: The scripted events are unchanged.
If the script is inaccurate, make a new PR for script changes.

**Target branch(es):** 3.3.5 / 6.x (both)

**Issues addressed:** Updates https://github.com/velinath/TrinityCore/issues/2 ([**Gossip migration from core scripts to SAI/DB ToDo List**](https://github.com/velinath/TrinityCore/issues/2))

**Tests performed:** (Does it build, tested in-game, etc) Builds witout issues.

**Known issues and TODO list:**
- [X] find actual gossip_menu_id value in sniff from retail (classic MC is available without raid group)
  (current values used in this PR comes from the 6.x branch TDB 6.04)

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp [hunk 1]
```diff
@@ -192,8 +196,8 @@ class boss_majordomo : public CreatureScript
 
         bool OnGossipHello(Player* player, Creature* creature) override
         {
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_SELECT, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
-            player->SEND_GOSSIP_MENU(GOSSIP_HELLO, creature->GetGUID());
+            player->ADD_GOSSIP_ITEM_DB(MENU_OPTION_TELL_ME_MORE, OPTION_ID_TELL_ME_MORE, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
+            player->SEND_GOSSIP_MENU(NPC_TEXT_THE_FIRELORD, creature->GetGUID());
             return true;
```
Comment (joschiwald) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp]: remove this gossip hook, since it is handled via db

Comment (tkrokli) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp]: OK, will do.

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp [hunk 2]
```diff
@@ -190,13 +187,6 @@ class boss_majordomo : public CreatureScript
             }
         };
 
-        bool OnGossipHello(Player* player, Creature* creature) override
-        {
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_SELECT, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
-            player->SEND_GOSSIP_MENU(GOSSIP_HELLO, creature->GetGUID());
-            return true;
-        }
```
Comment (joschiwald) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp]: now it won't work, you also have to change OnGossipSelect to sGossipSelect

**Comment by tkrokli** [comment]
I know that it does not compile yet, but I had to try to implement `sGossipSelect()` sometime soon.
I would appreciate some helpful pointers. I could not even make it work with `override` at the end.

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp [hunk 3]
```diff
@@ -188,21 +191,16 @@ class boss_majordomo : public CreatureScript
...
-
-        bool OnGossipSelect(Player* player, Creature* creature, uint32 /*sender*/, uint32 /*action*/) override
-        {
-            player->CLOSE_GOSSIP_MENU();
-            creature->AI()->DoAction(ACTION_START_RAGNAROS);
-            return true;
-        }
+            void sGossipSelect(Player* player, Creature* creature, uint32 menuId, uint32 gossipListId)
```
Comment (joschiwald) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp]: wrong args, that's why override fails at this point, that's why there is a override keyword

**Comment by tkrokli** [comment]
Necessary changes applied for the PR to build successfully in Travis.

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp (hunk 2 above)
Comment (tkrokli) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp]: Fixed ≈ 36 hours ago.

**Comment by tkrokli** [comment]
Inserted gossip_menu_id + gossip_menu_option values reportedly used in unofficial 4.3.4 branch.
Added `FACTION_FRIENDLY = 35` to enum to replace 2 occurrences of "magic numbers" in code.

**Comment by tkrokli** [comment]
Suggestion: maybe add the labels [Comp-C++Script] + [Comp-Database] to this PR?

**Comment by Aokromes** [comment]
Any reason to don't merge this?

**Comment by tkrokli** [comment]
According to joschiwald, the gossip_menu_id values still need verification from sniffs. It is doable, but I have not had time or chance yet to enter Molten Core again on retail as level 90+ to sniff them. (I did make an attempt ~2 months ago, but I failed to follow Majordomo Executus to Ragnaros' spawn point because the current TC script was misleading and made me think that it would take place at Majordomo's loot crate.) If anyone else has time to sniff the gossip options from Majordomo Executus, it would be a great help to me and to make this PR 100% ready for merging.

**Comment by Aokromes** [comment]
DELETE FROM gossip_menu WHERE (`entry`=4108 AND `text_id`=5012) OR (`entry`=4109 AND `text_id`=5011) OR (`entry`=4093 AND `text_id`=4995);
INSERT INTO gossip_menu (`entry`, `text_id`) VALUES
(4108, 5012), -- 54404
(4109, 5011), -- 54404
(4093, 4995); -- 54404
DELETE FROM gossip_menu_option WHERE (`menu_id`=4108 AND `id`=0) OR (`menu_id`=4109 AND `id`=0) OR (`menu_id`=4093 AND `id`=0);
INSERT INTO gossip_menu_option (`menu_id`, `id`, `option_icon`, `option_text`, `OptionBroadcastTextID`, `option_id`, `npc_option_npcflag`, `action_menu_id`, `action_poi_id`, `box_coded`, `box_money`, `box_text`, `BoxBroadcastTextID`) VALUES
(4108, 0, 0, 'You challenged us and we have come. Where is this master you speak of?\n', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 4108
(4109, 0, 0, 'What else do you have to say?', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN), -- 4109
(4093, 0, 0, 'Tell me more.', UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, 0, 0, '', UNKNOWN); -- 4093

**Comment by tkrokli** [comment]
Thank you for the added information, Aokromes. At least now I know that the same `gossip_menu_option`.`menu_id` values are used in the 6.x branch TDB 6.04 already.

**Pull request merged**